res/res_respoke/respoke_endpoint.c

```diff
@@ -270,59 +350,64 @@ static int respoke_endpoint_apply(const struct ast_sorcery *sorcery, void *obj)
 	}
 
 	/* Determine if we need fresh state and to reconnect */
-	if (endpoint->state) {
-		if (can_reuse_state(endpoint->state, app, transport)) {
+	state = respoke_endpoint_state_retrieve(ast_sorcery_object_get_id(endpoint));
+	if (state) {
+		if (can_reuse_state(state, app, transport)) {
 			/* Nothing has changed that warrants us reconnecting */
 			ao2_ref(app, -1);
 			ao2_ref(transport, -1);
 			return 0;
 		}
-		ao2_ref(endpoint->state, -1);
+		/* Can't reuse it; get rid of the old state */
+		ao2_unlink(states, state);
+		ao2_ref(state, -1);
+		state = NULL;
```

Comment:
Hrm yeah I guess I could see either way depending on the use case. Sounds like in this situation throwing it away would be fine.